geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/vlm/RasterSource.scala

```diff
@@ -150,6 +270,125 @@ object RasterSource {
+    layerType: String,
+    rdd: RDD[Seq[(String, String)]],
+    layoutType: LayoutType,
+    targetCRS: String,
+    numPartitions: Option[Int],
+    resampleMethod: ResampleMethod,
+    readMethod: String
+  ): SpatialTiledRasterLayer = {
+    val rasterSourcesRDD: RDD[Seq[(String, RasterSource)]] =
+      (readMethod match {
+        case GEOTRELLIS =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, new GeoTiffRasterSource(v): RasterSource)
+              }
+            }
+          }
+        case GDAL =>
+          rdd.mapPartitions { iter =>
+            iter.map { files: Seq[(String, String)] =>
+              files.map { case (k, v) => (k, GDALRasterSource(v): RasterSource)
+              }
+            }
+          }
+      }).cache()
+
+    val reprojectedSourcesRDD: RDD[(String, RasterSource)] =
+      targetCRS match {
+        case crs: String =>
+          rasterSourcesRDD.flatMap { case sources =>
+            sources.map { case (index, source) =>
+              (index, source.reproject(CRS.fromString(crs), resampleMethod))
+            }
+          }
+        case null => rasterSourcesRDD.flatMap { sources => sources }
+      }
+
+    val rasterSummary: RasterSummary =
+      reprojectedSourcesRDD
+        .map { case (_, source) =>
+            val ProjectedExtent(extent, crs) = source.getComponent[ProjectedExtent]
+            val cellSize = CellSize(extent, source.cols, source.rows)
+            RasterSummary(crs, source.cellType, cellSize, extent, source.size, 1)
+          }.reduce { _ combine _ }
+
+    val LayoutLevel(zoom, layout) =
+      layoutType match {
+        case global: GlobalLayout =>
+          val scheme = ZoomedLayoutScheme(rasterSummary.crs, global.tileSize)
+          scheme.levelForZoom(global.zoom)
+        case local: LocalLayout =>
+          val scheme = FloatingLayoutScheme(local.tileCols, local.tileRows)
+          rasterSummary.levelFor(scheme)
+      }
+
+    val tileLayerMetadata: TileLayerMetadata[SpatialKey] =
+      rasterSummary.toTileLayerMetadata(layout, zoom)._1
+
+    val layoutsRDD: RDD[(String, LayoutTileSource)] =
+      reprojectedSourcesRDD.mapValues { _.tileToLayout(layout, resampleMethod) }
+
+    val rasterRegionRDD: RDD[(SpatialKey, (String, RasterRegion))] =
+      layoutsRDD.flatMap { case (index, source) =>
+        source.keyedRasterRegions().map { case (key, region) =>
+          (key, (index, region))
+        }
+      }
+
+    val groupedRDD: RDD[(SpatialKey, Iterable[(String, RasterRegion)])] =
+      rasterRegionRDD
+        .groupByKey(SpatialPartitioner(numPartitions.getOrElse(rasterSummary.estimatePartitionsNumber)))
+
+    val tiledRDD: RDD[(SpatialKey, MultibandTile)] =
+      groupedRDD.mapValues { iter =>
+        val sorted = iter.toSeq.sortBy { _._1 }
+
+        MultibandTile(sorted.flatMap { case (_, ref) => ref.raster.toSeq.flatMap { _.tile.bands } })
```

Comment:
As previously mentioned you can't rely on the fact that a tile for every band is present, therefore `sort` is insufficient. Logic needs to be added to insert `NODATA` tiles for those bands which happen to be missing.